src/lib/php/Application/RepositoryApi.php

```diff
@@ -16,42 +16,95 @@
 
 namespace Fossology\Lib\Application;
 
-class RepositoryApi {
+class CurlRequest
+{
+  private $handle = null;
+
+  public static function create($url)
+  {
+    return new CurlRequest($url);
+  }
+
+  public function __construct($url)
+  {
+    $this->handle = curl_init($url);
+  }
+
+  public function setOptions($options)
+  {
+    curl_setopt_array($this->handle, $options);
+  }
+
+  public function execute()
+  {
+    return curl_exec($this->handle);
+  }
+
+  public function getInfo($resource)
+  {
+    return curl_getinfo($this->handle, $resource);
+  }
+
+  public function close()
+  {
+    curl_close($this->handle);
+  }
+}
+
+class CurlRequestServer
+{
+  public function create($url)
+  {
+    return new CurlRequest($url);
+  }
+}
+
+class RepositoryApi
```

Comment:
We usually do not have multiple classes in one file

Reply:
Done.